adcc/testdata/qchem/generate_qchem_data.py

```diff
@@ -0,0 +1,171 @@
+    "def2tzvp": "def2-tzvp",
+    "ccpvdz": "cc-pvdz",
+}
+
+
+def clean_xyz(xyz):
+    return "\n".join(x.strip() for x in xyz.splitlines())
+
+
+def generate_qchem_input_file(fname, method, basis, coords, potfile=None,
+                              charge=0, multiplicity=1, memory=8000,
+                              singlet_states=5, triplet_states=0, bohr=True,
+                              maxiter=160, conv_tol=10, n_core_orbitals=0):
+    nguess_singles = 2 * max(singlet_states, triplet_states)
+    max_ss = 5 * nguess_singles
+    pe = potfile is not None
+    qci = _qchem_template.format(
+        method=_method_dict[method],
+        basis=basis_remap[basis],
+        memory=memory,
+        singlet_states=singlet_states,
+        triplet_states=triplet_states,
+        n_guesses=nguess_singles,
+        bohr=bohr,
+        maxiter=maxiter,
+        conv_tol=conv_tol,
+        max_ss=max_ss,
+        charge=charge,
+        multiplicity=multiplicity,
+        cc_rest_occ=n_core_orbitals,
+        xyz=clean_xyz(coords),
+        pe=pe
+    )
+    if pe:
+        qci += _pe_template.format(potfile=potfile)
+    qc_file = open(fname, "w")
+    qc_file.write(qci)
+    qc_file.close()
+
+
+def dump_qchem(molecule, method, basis, **kwargs):
+    with tempfile.TemporaryDirectory() as tmpdir:
+        pe = kwargs.get("potfile", None) is not None
+        if pe:
+            basename = f"{molecule}_{basis}_pe_{method}"
+        else:
+            basename = f"{molecule}_{basis}_{method}"
+        infile = os.path.join(tmpdir, basename + ".in")
+        outfile = os.path.join(tmpdir, basename + ".out")
+        geom = xyz[molecule].strip()
+        ret = {
+            "molecule": molecule,
+            "method": method,
+            "basis": basis,
+            "pe": pe,
+            "geometry": geom
+        }
+        generate_qchem_input_file(
+            infile, method, basis, geom, **kwargs
+        )
+        # only works with my (ms) fork of cclib
+        # github.com/maxscheurer/cclib, branch dev-qchem
+        # os.system("qchem {} {}".format(infile, outfile))
+        sh.qchem(infile, outfile)
+        res = QChem(outfile).parse()
+        ret["oscillator_strengths"] = res.etoscs
+        ret["state_dipole_moments_debye"] = res.etdipmoms
+        ret["excitation_energies_ev"] = res.etenergies
+        if pe:
+            ret["pe_ptss_corrections_ev"] = res.peenergies["ptSS"]
+            ret["pe_ptlr_corrections_ev"] = res.peenergies["ptLR"]
+        return basename, ret
+
+
+def main():
+    basissets = ["sto3g", "ccpvdz"]
+    methods = ["adc1", "adc2", "adc3"]
+    qchem_results = {}
+    for method in methods:
+        for basis in basissets:
+            key, ret = dump_qchem("formaldehyde", method, basis,
+                                  potfile=pe_potentials["fa_6w"])
+            qchem_results[key] = ret
+            print(f"Dumped {key}.")
+    hdf5io.save(fname="qchem_dump.hdf5", dictionary=qchem_results)
```

Comment:
Since your hdf5 file actually only contains a few scalar datapoints (and no arrays like in the other cases) I could be tempted to just check it into the repository as a yaml file. That saves a lot of hassle with the downloading etc.

Also: Is there any reason for this to live in a subfolder? So far all generation scripts have lived in the main `testdata` folder. You could of course dump the qchem input and output into a subdirectory, but that does not need to be checked in.

Reply:
Also I would dump the data as close as possible to the structure of the other datafiles (i.e. use bohr units, same hierarchy / names of keys for the properties etc. That makes it easier to mix match and extend as needed.

Reply:
I'm for the yaml solution. How would we deal with the `np` arrays?
I guess `tolist`, otherwise we get binary stuff in there as well.

Reply:
Yes I think that is the best option. There is also the "proper" way to use representers, see [molsturm](https://github.com/molsturm/molsturm/blob/master/src/interface/python/molsturm/yaml_utils.py), but I'm not sure that is appropriate here. If we only need it once it's a bit over the top.

Reply:
I'd say we add the representers... Can probably be used if we want to serialized `ExcitedStates` to yaml? We could create a `yamlio.py`
(with the user-exposed functions `save` and `load`)
analogous to `hdf5io.py`? I think I'd enjoy having that 😄

Reply:
Well I'm no longer a fan of hdf5io, actually. It's really not such a well-written piece of code and especially it is not perfectly compatible to the rest of the world. That same problem we have if we use our custom representers for yaml, but ok. If you think it helps, than go for it.

Reply:
The "outside" world argument is a good point. I guess I'll convert the arrays to lists, and convert back to numpy array upon read-in. Then we add less code, which is kind of good.

Reply:
Done.